shared/java/com/google/idea/blaze/common/Label.java

```diff
@@ -57,17 +56,22 @@ public static Label of(String label) {
   }
 
   public static Label fromWorkspacePackageAndName(String workspace, Path packagePath, Path name) {
-    final var systemIndependentPackagePath = StreamSupport.stream(packagePath.spliterator(), false)
-        .map(Path::toString)
-        .collect(Collectors.joining("/"));
-
+    String packageWithForwardSlashes = withForwardSlashes(packagePath);
+    String nameWithForwardSlashes = withForwardSlashes(name);
     if (workspace.isEmpty()) {
-      return Label.of(String.format("//%s:%s", systemIndependentPackagePath, name));
+      return Label.of(String.format("//%s:%s", packageWithForwardSlashes, nameWithForwardSlashes));
     } else {
-      return Label.of(String.format("@@%s//%s:%s", workspace, systemIndependentPackagePath, name));
+      return Label.of(String.format("@@%s//%s:%s", workspace, packageWithForwardSlashes, nameWithForwardSlashes));
     }
   }
 
+  private static String withForwardSlashes(Path p) {
+    StringBuilder res = new StringBuilder();
+    p.iterator().forEachRemaining(part -> res.append(part).append("/"));
+    // Remove the trailing slash
+    return res.substring(0, res.length() - 1);
```

Comment:
Blows up for empty path.

Reply:
I don't think this can ever happen (even `Path.of("")` returns an iterator that is not empty), but I've added a check just in case.

Reply:
Oh, yes looks like you are right.